contentcuration/contentcuration/frontend/channelEdit/pages/StagingTreePage/index.vue

```diff
@@ -591,6 +642,25 @@
 
         this.deployCurrentChannel();
       },
+      onPublishDraftClick() {
+        this.displayPublishDraftDialog = false;
+        this.isPublishingDraft = true;
+
+        this.publishDraftChannel()
```

Comment:
The only slight issue here is that this is only waiting on the change having been created - not that it has been applied on the backend. We would need to add similar logic to the `waitForDeploying` (although my recent experience with this suggests it is not working as intended).